src/record/api.rs

```diff
@@ -706,6 +724,20 @@ mod tests {
     }};
   }
 
+  // NOTE:
```

Comment:
I think we should refactor our tests, there is a lot of code duplication here and there.